src/doc/examples/lock-nodes.md

```diff
@@ -0,0 +1,184 @@
+# Examples
+
+## Node depended resources
+
+Create resources that are node depended is in many cases very helpful. That means a job must pick a target node where the requested resource is available.
+
+```groovy
+// allocate node
+node('some-build-node') {
+  // Lock resource named *whatever-resource-some-build-node*
+  lock("whatever-resource-${env.NODE_NAME}") {
+    echo "I am no node ${env.NODE_NAME} and locked resource ${env.LOCKED_RESOURCE}"
+  }
+}
+```
+
+But much more useful is lock node first.
```

Comment:
This example does not acquire the lock on `whatever-resource-${env.NODE_NAME}"`.  Shouldn't it acquire that lock in order to be consistent with the earlier example?
```suggestion
But it is much more useful to lock the node before locking the node-specific resource.
```